src/Server/modules/USPlacesManager.js

```diff
@@ -0,0 +1,340 @@
+const dataClassesModule = require('../modules/DataClasses.js');
+const readCSVModule = require('../modules/ReadCSV.js');
+const writeCSVModule = require('../modules/WriteCSV.js');
+//const getRecentModule = require('../modules/World');
+
+var allUSPlaces = [];
+
+function InitializeAllUSPlace(){
+    allUSPlaces = readCSVModule.LoadUSData();
+}// read allUSdata and put into array
+
+function AddUSData(county, state, date, tod, number) {
+    
+    if(allUSPlaces == undefined){
+        InitializeAllUSPlace();
+    }
+
+	var errormsg = 'no error';
+	if (tod == 1) {
+		for (var i = 0; i < allUSPlaces.length; ++i) {
+			if (county == allUSPlaces[i].county && state == allUSPlaces[i].state) {
+				for (var key in allUSPlaces[i].numConfirmed) {
+					var temp_date = new Date(key);
+					var d =
+						temp_date.getMonth() +
+						1 +
+						'/' +
+						temp_date.getDate() +
+						'/' +
+						temp_date.getFullYear();
+					if (d == date) {
+						console.log('ERROOOOOOOR');
+						//return result;
+						errormsg = 'date exists';
+						return errormsg;
+					}
+				}
+				var temp_date = helper.stringToDate(date);
+				allUSPlaces[i].numDeaths[temp_date] = number;
+				if(allUSPlaces[i].numDeaths[temp_date] == GetMostRecentValue(allUSPlaces[i].numDeaths)){
+					allUSPlaces[i].currentNumDeaths=number;
+				}//check if the number is the most recent number
+				break;
+			} else if (i == allUSPlaces.length - 1) {
+				// if its the last element & prog didn't go into the 1st if -> place doesn't exist
+				errormsg = 'wrong place';
+			}
+		}
+	} else if (tod == 2) {
+		for (var i = 0; i < allUSPlaces.length; ++i) {
+			if (county == allUSPlaces[i].county && state == allUSPlaces[i].state) {
+				for (var key in allUSPlaces[i].numDeaths) {
+					var temp_date = new Date(key);
+					var d =
+						temp_date.getMonth() +
+						1 +
+						'/' +
+						temp_date.getDate() +
+						'/' +
+						temp_date.getFullYear();
+					if (d == date) {
+						console.log('ERROOOOOOOR');
+						errormsg = 'date exists';
+						return errormsg;
+					}
+				}
+				var temp_date = helper.stringToDate(date);
+				allUSPlaces[i].numConfirmed[temp_date] = number;
+				if(allUSPlaces[i].numConfirmed[temp_date] == GetMostRecentValue(allUSPlaces[i].numConfirmed)){
+					allUSPlaces[i].currentNumConfirmed=number;
+				}
+				break;
+			} else if (i == allUSPlaces.length - 1) {
+				// if its the last element & prog didn't go into the 1st if -> place doesn't exist
+				errormsg = 'wrong place';
+				break; //just in case
+			}
+		}
+	}
+	
+
+	return errormsg;
+}
+
+function EditUSData(county, state, date, tod, number) {
```

Comment:
We need to update currentNumConfirmed and currentNumDeaths inside this function, just like inside AddUSData().